### ISSUE TITLE
tqdm integration ( Progress Bar )

### ISSUE THREAD

**Issue by IlfirinPL** [robotframework/robotframework#4376]
I have question did its possible to integrate tqdm library to loops in robotframework ( maybe some additional flag tqdm=True

Some loops take long 
Example Code
```
    FOR    ${i}    IN RANGE    30    tqdm=True
        Log    ${i}
        Sleep    1
    END

```
also for long test case it would be nice to make process bar for user

```
Start TQDM    100
Update  TQDM    10
Update  TQDM    1

``` 

https://github.com/tqdm/tqdm

**Comment by pekkaklarck** [comment]
Better console output would in general be nice, but there's a problem that we don't want RF core to have mandatory external dependencies. The motivation for that decision is that RF is often installed in places not connected to the Internet (and PyPI) and just having a single installation package makes things a lot easier in cases like that.

With FOR loops I'd be also very concerned about how much output there would be if you have a fast loop with lot of iterations. Possibly this could be configured somehow, but adding new syntax to the data for that purpose doesn't sound good to me.

What could be done instead is creating a listener creates a custom console output with whatever features needed. Not being part of the core would have several benefits:
- No restriction on using third-party modules.
- Can be developed on its own pace without being locked to core's somewhat infrequent release schedule.
- No need for as strict backwards compatibility as with the core.

In other words, I don't think this would be a good feature in the core, but I'd love to see something like this being available as an extension.

**Comment by IlfirinPL** [comment]
I see wisdom of this approach.
